### ISSUE TITLE
Allow app consumers to set h5py options

### ISSUE THREAD

**Issue by loichuder** [silx-kit/h5grove#80]
We are using h5grove (in our fastapi project) to serving the h5 file to our frontend, just wondering is it possible to pass h5py optins like swmr=True with the endpoint? cause now the get_content_from_file and get_list_of_paths do accept the h5py options, but we can't pass in from the endpoint or the settings.

_Originally posted by @Tinanuaa in https://github.com/silx-kit/h5grove/issues/73#issuecomment-1622934407_

**Comment by loichuder** [comment]
Valid point.

I reckon having general settings for `h5py` in the app would make more sense than setting it at each request.

**Comment by Tinanuaa** [comment]
I agree, I would prefer to have some general settings  like  swmr=True that  we can set through the fastapi_utils.settings
